apps/src/code-studio/pd/professional_learning/RegionalWorkshopCatalog.jsx

```diff
@@ -41,8 +41,14 @@ export default function RegionalWorkshopCatalog({
   // the "National workshops" section since they'll show up under the
   // "Upcoming local workshops" section.
   const availableNationalWorkshops = useMemo(() => {
-    return nationalWorkshops.filter(
-      ws => !availableRegionalWorkshops.includes(ws.id)
+    if (!availableRegionalWorkshops) {
+      return nationalWorkshops;
+    }
+    const availableRegionalWorkshopIds = availableRegionalWorkshops.map(
+      ws => ws.id
+    );
+    return nationalWorkshops?.filter(
+      ws => !availableRegionalWorkshopIds.includes(ws.id)
```

Comment:
```suggestion
    const availableRegionalWorkshopIds = new Set(availableRegionalWorkshops.map(
      ws => ws.id
    ));
    return nationalWorkshops?.filter(
      ws => !availableRegionalWorkshopIds.has(ws.id)
```

while the size of both of these arrays are likely pretty small, this can cause an exponential performance bottleneck in larger arrays. using a `Set` instead of an array means the lookup step doesn't need to iterate through the `availableRegionalWorkshopIds` to determine if the id exists. might be an over-optimization here but it's a good habit to get into